### DISCUSSION TITLE
How to push an index to text array with supabase js library

### DISCUSSION THREAD

**Discussion by saman2007** [supabase/supabase#5998] [Questions]
Hello. I have a question. To add an index to a text array, my soloution is: 
1- get the text array that you want to push an item to
2- push the item to the array you got from supabase
3- update the colomn that you got the text array from there

But is there an other way to push the item to text array?
For example pushing the item to the text array using supabase js library.
Like this: i pass my item as an argument to supabase library and supabase will push that item to the text array that i mentioned.

Is it possible? If yes, please tell me how. Thanks a lot.

**Comment by GaryAustin1** [comment]
There is no direct API way available.  
I would use an rpc call passing the item to a Postgres function and do:
UPDATE table SET array_field = array_append(array_field,'item') WHERE ...
In the function. 
Postgres array functions https://www.postgresql.org/docs/current/functions-array.html
